suites/algorithms.js

```diff
@@ -0,0 +1,134 @@
+import chai from 'chai';
+import {createInitialVc} from '../helpers.js';
+
+const expect = chai.expect;
+
+export function algorithmsAssertions(
+  {
+    endpoints, testDescription,
+    vendorName, credential = {
+      dates: true,
+      contextInjection: true,
+      domain: true
+    }
+  }) {
+  return describe(testDescription, function() {
+    const columnId = testDescription;
+    const [issuer] = endpoints;
+    beforeEach(function() {
+      this.currentTest.cell = {
+        columnId,
+        rowId: this.currentTest.title
+      };
+    });
+    let proofs = [];
+    let data;
+    before(async function() {
+      if(!issuer) {
+        throw new Error(`Expected ${vendorName} to have an issuer.`);
+      }
+      if(!credential) {
+        throw new Error(`Expected a credential to be passed in ` +
+              `received ${credential}`);
+      }
+      data = await createInitialVc({issuer, credential});
+      proofs = Array.isArray(data.proof) ? data.proof : [data.proof];
+    });
+    it('Whenever this algorithm encodes strings, it MUST use UTF-8 encoding.',
+      function() {
+        this.test.link = 'https://www.w3.org/TR/vc-data-integrity/#:~:text=or%20an%20error.-,Whenever%20this%20algorithm%20encodes%20strings%2C%20it%20MUST%20use%20UTF%2D8%20encoding.,-Let%20proof%20be';
+        for(const proof of proofs) {
+          expect(proof.proofValue.isWellFormed()).to.be.true;
+        }
+      });
+    it('If the algorithm produces an error, the error MUST be propagated \
+and SHOULD convey the error type.',
+    function() {
+      this.test.link = 'https://www.w3.org/TR/vc-data-integrity/#:~:text=If%20the%20algorithm%20produces%20an%20error%2C%20the%20error%20MUST%20be%20propagated%20and%20SHOULD%20convey%20the%20error%20type.';
+      this.test.cell.skipMessage = 'Pending test.';
+      this.skip();
+    });
+    it('If one or more of the proof.type, proof.verificationMethod, \
+and proof.proofPurpose values is not set, an error MUST be raised.',
+    function() {
+      this.test.link = 'https://www.w3.org/TR/vc-data-integrity/#:~:text=If%20one%20or%20more%20of%20the%20proof.type%2C%20proof.verificationMethod%2C%20and%20proof.proofPurpose%20values%20is%20not%20set%2C%20an%20error%20MUST%20be%20raised%20and%20SHOULD%20convey%20an%20error%20type%20of%20PROOF_GENERATION_ERROR.';
+      for(const proof of proofs) {
+        expect(proof).to.contain.keys(
+          'type', 'proofPurpose', 'verificationMethod');
+      }
+    });
+    it('If options has a non-null domain item, it MUST be equal to \
```

Comment:
there is a verifier test for this I believe if there isn't we usually test this stuff in the verifier suite.